python/requirements-wheel-test.txt

```diff
@@ -5,22 +5,8 @@ pytest
 pytz
 tzdata; sys_platform == 'win32'
 
-numpy==1.21.3; platform_system == "Linux"   and platform_machine == "aarch64" and python_version < "3.11"
-numpy==1.23.4;                                                                    python_version == "3.11"
-numpy==1.26.0;                                                                    python_version >= "3.12"
-numpy==1.19.5; platform_system == "Linux"   and platform_machine != "aarch64" and python_version <  "3.9"
-numpy==1.21.3; platform_system == "Linux"   and platform_machine != "aarch64" and python_version >= "3.9" and python_version < "3.11"
-numpy==1.21.3; platform_system == "Darwin"  and platform_machine == "arm64"   and python_version <  "3.11"
-numpy==1.19.5; platform_system == "Darwin"  and platform_machine != "arm64"   and python_version <  "3.9"
-numpy==1.21.3; platform_system == "Darwin"  and platform_machine != "arm64"   and python_version >= "3.9" and python_version < "3.11"
-numpy==1.19.5; platform_system == "Windows"                                   and python_version <  "3.9"
-numpy==1.21.3; platform_system == "Windows"                                   and python_version >= "3.9" and python_version < "3.11"
+numpy~=1.21.3; python_version < "3.11"
+numpy~=1.23.2; python_version == "3.11"
+numpy~=1.26.0; python_version == "3.12"
```

Comment:
How is `~=` different from `==` in requirements file?

Reply:
Also, should be include a line for Python 3.13?

Reply:
> How is `~=` different from `==` in requirements file?

I was initially using `==` as before, but after reading https://packaging.python.org/en/latest/specifications/version-specifiers/#id5 I thought this was the best option (although I would also have to look it up again, which makes it not ideal maintenance-wise, maybe I should add a comment explaining the logic). 
The `~=` is called a "compatible" release, and essentially `numpy~=1.26.0` translates to `>= 1.26.0, == 1.26.*`. Using this instead of `==` avoids us bumping those exact pins to the latest bug-fix release in that x.y series, as we sometimes did in the past ([example](https://github.com/apache/arrow/commit/721aa019aa0cb3b6a15aa358a53f39e2c5c541ae#diff-3385c98e15ee29eedc051bdefb0755af30f4d882ff0038ce1a46f677c505ac78))

> Also, should be include a line for Python 3.13?

I had removed that again because it is being added in https://github.com/apache/arrow/pull/43539 (anyway, depending on which PR is merged first, the other will have to fix the conflicts for this file)

Reply:
Added a comment